attending/downloader.py

```diff
@@ -0,0 +1,38 @@
+import urllib.request
+from pathlib import Path
+from zipfile import ZipFile
+
+from .Doc import DocLocation
+
+
+def extract_zip(doc_location: DocLocation, file: Path):
+    ZipFile(file).extractall(path=doc_location.full_path())
+    file.unlink()
+
+
+def write_to_file(url, doc_location: DocLocation):
+    with urllib.request.urlopen(url) as connection:
+        if 400 > connection.status >= 300:
+            write_to_file(connection.getheader("Location"), doc_location)
+        elif connection.status == 200:
+            file_extension = _EXTENSION_MAPPING.get(connection.getheader('Content-Type'), 'txt')
+            target = doc_location.full_path() / Path(f"{doc_location.name}.{file_extension}")
```

Comment:
Might we want to preserve the original filename, if there was one, from the url?

Reply:
yes, in fact. the example `PDF` file that I am using for `IPython` does exactly that.